langchain/document_transformers/interrogate.py

```diff
@@ -0,0 +1,32 @@
+"""Transform documents"""
+from typing import Any, Sequence
+
+from pydantic import BaseModel
+
+from langchain.utils import get_from_dict_or_env
+from langchain.schema import BaseDocumentTransformer, Document
+from doctran import Doctran
+
+
+class DocumentInterrogator(BaseDocumentTransformer, BaseModel):
```

Comment:
It should be minimal changes to make this a `parser`. 

1/ Inherit from `BaseBlobParser`
2/ Just re-name `atransform_documents` to `lazy_parse` (as shown [here](https://github.com/hwchase17/langchain/blob/2eb4a2ceeab8fb2e7b64461c71e10c468ff0ca01/langchain/document_loaders/parsers/audio.py#L15C5-L15C8)) to accept blobs
3/ should then work out-of-the-box w/ the blob loaders

The UX in the notebook is then simple:

```
loader = GenericLoader.from_filesystem(
    "./example_data/pdf",
    glob="*",
    suffixes=[".pdf"],
    parser= QA_generator()
)
docs = loader.load()
```